explicitly provide memory format when calling to clone() at ProcessGroupGloo.cpp

ifedan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#28688 explicitly provide memory format when calling to clone() at ProcessGroupGloo.cpp**

Differential Revision: [D18333382](https://our.internmc.facebook.com/intern/diff/D18333382)

mrshenli commented on torch/lib/c10d/ProcessGroupGloo.cpp:
```diff
@@ -1017,7 +1017,7 @@ class AsyncSparseAllreduceWork : public ProcessGroupGloo::AsyncWork {
     outputs.reserve(inputs.size());
     for (size_t i = 0; i < inputs.size(); i++) {
-      outputs.push_back(output.clone());
+      outputs.push_back(output.clone(at::MemoryFormat::Contiguous));
```
@ifedan What does this mean for sparse tensors?

ifedan committed:
Update on "explicitly provide memory format when calling to clone() at ProcessGroupGloo.cpp"


Differential Revision: [D18333382](https://our.internmc.facebook.com/intern/diff/D18333382)

[ghstack-poisoned]

ifedan committed:
Update on "explicitly provide memory format when calling to clone() at ProcessGroupGloo.cpp"


Differential Revision: [D18333382](https://our.internmc.facebook.com/intern/diff/D18333382)

[ghstack-poisoned]

ifedan committed:
Update on "explicitly provide memory format when calling to clone() at ProcessGroupGloo.cpp"


Differential Revision: [D18333382](https://our.internmc.facebook.com/intern/diff/D18333382)

[ghstack-poisoned]

ifedan committed:
Update on "explicitly provide memory format when calling to clone() at ProcessGroupGloo.cpp"


Differential Revision: [D18333382](https://our.internmc.facebook.com/intern/diff/D18333382)

[ghstack-poisoned]
